daily_milestone.go

```diff
@@ -1,29 +1,225 @@
+	StartDate   string     `json:"start_date"`
+	DueDate     string     `json:"due_date"`
+}
+type gitLabAPI struct {
+	ID          int        `json:"id"`
+	Iid         int        `json:"iid"`
+	ProjectID   int        `json:"project_id"`
+	Title       string     `json:"title"`
+	Description string     `json:"description"`
+	StartDate   string     `json:"start_date"`
+	DueDate     string     `json:"due_date"`
+	State       string     `json:"state"`
+	UpdatedAt   *time.Time `json:"updated_at"`
+	CreatedAt   *time.Time `json:"created_at"`
+	Name        string     `json:"name"`
+	NameSpace   struct {
+		ID       int    `json:"id"`
+		Name     string `json:"name"`
+		Path     string `json:"path"`
+		Kind     string `json:"kind"`
+		FullPath string `json:"full_path"`
+	} `json:"namespace"`
+}
+
+// Initialization of logging variable
+var logger *log.Logger
+
+// LoggerSetup Initialization
+func LoggerSetup(info io.Writer) {
+	logger = log.New(info, "INFO: ", log.Ldate|log.Ltime|log.Lshortfile)
+}
+
+// Function to get project ID from the gitLabAPI
+func getProjectID(baseURL string, Token string, Projectname string, Namespace string) (string, error) {
+	project := []gitLabAPI{}
+	urls := "https://" + baseURL + "/api/v4" + "/projects"
+	client := &http.Client{}
+	req, err := http.NewRequest("GET", urls, nil)
+	if err != nil {
+		logger.Println(err)
+
+	}
+	req.Header.Add("PRIVATE-TOKEN", Token)
+	resp, err := client.Do(req)
+	if err != nil {
+		logger.Println(err)
+	}
+
+	respByte, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		logger.Println("fail to read response data")
+
+	}
+	json.Unmarshal(respByte, &project)
+	defer resp.Body.Close()
+	for _, p := range project {
+		if p.Name == "message" {
+			fmt.Println(p.Name)
+		}
+
+		if p.Name == Projectname && p.NameSpace.Path == Namespace {
+			fmt.Println(strconv.Itoa(p.ID))
+			return strconv.Itoa(p.ID), nil
+		}
+	}
+	return "", fmt.Errorf("project %s not found", Projectname)
+}
+
+// It is getting milestones data from the milestone API
+func getMilestones(baseURL string, token string, projectID string) ([]string, error) {
+	project := []milestoneAPI{}
+	list := []string{}
+	strurl := []string{"https://", baseURL, "/api/v4", "/projects/", projectID, "/milestones"}
+	urls := strings.Join(strurl, "")
+	client := &http.Client{}
+	req, err := http.NewRequest("GET", urls, nil)
+	if err != nil {
+		logger.Println(err)
 
-func get_project_id(base_url string, token string, projectname string, namespace string) string {
-	headers := map[string]string{"PrivateToken": token}
-	url := "https://" + base_url + "/projects"
-	page := 1
-	for {
+	}
+	req.Header.Add("PRIVATE-TOKEN", token)
+	resp, err := client.Do(req)
+	if err != nil {
+		logger.Println(err)
+
+	}
+	respByte, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		logger.Println("fail to read response data")
 
 	}
 
+	json.Unmarshal(respByte, &project)
+	defer resp.Body.Close()
+	for _, p := range project {
```

Comment:
Firstly project is singular, but we actually have a list again.
Secondly it doesn't seem to actually be a projectList, but a milestoneList. Why label it differently? Makes the code much harder to understand.